enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/NeoShallowEmbeddedInteraction.java

```diff
@@ -91,10 +91,12 @@ public void logout( EnterpriseAuthSubject subject )
         subject.logout();
     }
 
+    // To overcome that the Shiro subject caches it's authentication status.
+    // This assumes that zero authorization equals not being authenticated.
     @Override
     public boolean isAuthenticated( EnterpriseAuthSubject subject )
     {
-        return subject.getShiroSubject().isAuthenticated();
+        return subject.getShiroSubject().isAuthenticated() && subject.allowsReads();
```

Comment:
This assumption is a bit uncomfortable. Maybe we could check subject.getAuthenticationResult() (and make sure to keep it up-to-date)?